services/src/main/java/io/druid/cli/CliPeon.java

```diff
@@ -223,6 +231,34 @@ private void configureTaskActionClient(Binder binder)
                             .to(RemoteTaskActionClientFactory.class).in(LazySingleton.class);
 
           }
+
+          @Provides
+          @LazySingleton
+          public Task readTask(@Json ObjectMapper mapper, ExecutorLifecycleConfig config)
+          {
+            try {
+              return mapper.readValue(config.getTaskFile(), Task.class);
+            }
+            catch (IOException e) {
+              throw Throwables.propagate(e);
+            }
+          }
+
+          @Provides
+          @LazySingleton
+          @Named(MonitorsConfig.METRIC_DIMENSION_PREFIX + DruidMetrics.DATASOURCE)
+          public String getDataSourceFromTask(final Task task)
```

Comment:
Because its all about what is optional. I had a hard time figuring out how to make an optional binding in 4.0-beta (4.0 has [OptionalBinder](https://google.github.io/guice/api-docs/4.0/javadoc/com/google/inject/multibindings/OptionalBinder.html) which might work later) This is the best workaround I could come up with. Make an object bound everywhere, but the fields in the object be optional.